lib/Doctrine/ORM/Internal/Hydration/SimpleObjectHydrator.php

```diff
@@ -151,4 +155,23 @@ protected function hydrateRowData(array $sqlResult, array &$result)
             $this->_uow->hydrationComplete();
         }
     }
+
+    /**
+     * @param string $discriminator
+     * @param array $sqlResult
+     */
+    private function removeColumnsNotRelatedToEntityDiscriminator(string $discriminator, array &$sqlResult)
```

Comment:
Performance impact of this might be huge, and needs to be benchmarked against the performance test suite